networkx/readwrite/nx_ete.py

```diff
@@ -0,0 +1,221 @@
+
+    G = nx.DiGraph()
+
+    for tree_node in T.iter_search_nodes():
+        node_name = tree_node.name
+        node_features = {
+            k: getattr(tree_node, k) for k in tree_node.features - {"dist", "support"}
+        }
+
+        G.add_node(node_name, **node_features)
+
+    queue = deque([T])
+
+    while queue:
+        parent = queue.popleft()
+
+        for child in parent.children:
+            queue.append(child)
+            G.add_edge(parent.name, child.name)
+
+    return G
+
+
+@open_file(1, mode="w")
+def write_ete(G, path_for_ete_output):
+    """Write NetworkX arborescence [3]_ G in the New Hampshire Newick format.
+
+    ETE (Environment for Tree Exploration) is a Python programming toolkit
+    that assists in the automated manipulation,
+    analysis and visualization of phylogenetic trees [2]_.
+
+    It uses the New Hampshire Newick format as one of the most widely used
+    standard representation of trees in bioinformatics [1]_.
+
+
+    Parameters
+    ----------
+    G : NetworkX arborescence
+
+    path_for_ete_output : file or string
+       File or filename to write.
+
+    Examples
+    --------
+    >>> G = nx.path_graph(42, create_using=nx.DiGraph)
+    >>> nx.write_ete(G, "test_ete.txt")
+
+    References
+    ----------
+    .. [1] http://etetoolkit.org/docs/latest/tutorial/tutorial_trees.html#reading-and-writing-newick-trees
+    .. [2] http://etetoolkit.org/
+    .. [3] https://networkx.org/documentation/stable//reference/algorithms/tree.html#tree
+    """
+    tree = to_ete(G)
+
+    path_for_ete_output.write(
+        tree.write(
+            features=list(tree.features - {"dist", "support"}), format_root_node=True
+        )
+    )
+
+
+@open_file(0, mode="r")
+def read_ete(path):
+    """Read a NetworkX arborescence [3]_ in the New Hampshire Newick format.
+
+    ETE (Environment for Tree Exploration) is a Python programming toolkit
+    that assists in the automated manipulation,
+    analysis and visualization of phylogenetic trees [2]_.
+
+    It uses the New Hampshire Newick format as one of the most widely used
+    standard representation of trees in bioinformatics [1]_.
+
+    Parameters
+    ----------
+    path : file or string
+       File or filename to read.
+
+    Returns
+    -------
+    G : NetworkX arborescence
+
+    Examples
+    --------
+    >>> nx.write_ete(nx.path_graph(42, create_using=nx.DiGraph), "test_ete.txt")
+    >>> G = nx.read_ete("test_ete.txt")
+
+    References
+    ----------
+    .. [1] http://etetoolkit.org/docs/latest/tutorial/tutorial_trees.html#reading-and-writing-newick-trees
+    .. [2] http://etetoolkit.org/
+    .. [3] https://networkx.org/documentation/stable//reference/algorithms/tree.html#tree
+    """
+    try:
+        import ete3
+    except ImportError as e:
+        raise ImportError("to_ete() requires ete3: http://etetoolkit.org/") from e
+
+    file = [line for line in path]
+
+    tree = ete3.Tree(file[0])
```

Comment:
If only the first line is needed to construct the `ete3.Tree` object, it might be better (e.g. if there's a huge data file) to not construct the list from `file`, e.g. something like `tree = ete3.Tree(next(file))`. I'm not sure what ete3 expects though so maybe this wouldn't work...